docs/docs/user-setup-cli.mdx

```diff
@@ -0,0 +1,278 @@
+---
+title: (EXPERIMENTAL) Installing ZMK
+sidebar_label: (EXPERIMENTAL) Installing ZMK
+---
+
+This is an **experimental** command line program to help set up ZMK. [Click here to return to the main documentation](user-setup.mdx).
+
+ZMK CLI walks you through installing ZMK and setting up a GitHub repository to store and build custom firmware. It also automates some common tasks such as adding new keyboards to your repository.
+
+The instructions below contain commands that need to be run in a terminal program. On Windows, use the [Windows Terminal](https://apps.microsoft.com/detail/9n0dx20hk701) or PowerShell. On other operating systems, the terminal program is usually just named "Terminal".
+
+:::info
+If you use this program, any bug reports are highly appreciated. We are happy to help you fix your setup if you encounter a bug through the usage of this program.
+:::
+
+## Installation
+
+### Preliminaries
+
+#### Make sure you have Git installed
+
+```sh
+git --version
+```
+
+<details>
+  <summary>I don't have Git installed.</summary>
+
+Install Git from https://git-scm.com/downloads.
+
+If you have Windows 11, you can instead open a terminal and run:
+
+```sh
+winget install git.git
+```
+
+Re-open your terminal and test to make sure it is now installed.
+
+</details>
+
+#### Make sure you have Python installed
+
+```sh
+python --version
+python3 --version
+```
+
+ZMK CLI requires Python 3.10 or newer.
+
+<details>
+  <summary>I don't have Python 3.10 or newer installed.</summary>
+
+##### On Windows and macOS
+
+Install the latest version of Python from https://www.python.org/downloads/.
+
+If you have Windows 11, you can instead open a terminal and run:
+
+```sh
+winget install python3
+```
+
+Re-open your terminal and test to make sure it is now installed.
+
+##### On Linux
+
+Install `python3` with your package manager.
+
+If the version is older than 3.10, you will need to find and install a package for a newer version of Python. On Ubuntu 20.04 and older, you can get Python 3.10 from the deadsnakes PPA with the following commands:
+
+```sh
+sudo add-apt-repository ppa:deadsnakes/ppa
+sudo apt install python3.10
+```
+
+You will then need to replace `python3` with `python3.10` in the rest of the installation instructions.
+
+Re-open your terminal and test to make sure it is now installed.
+
+</details>
+
+#### (Recommended) Make sure you have pipx installed
+
+```sh
+pipx --version
+```
+
+<details>
+  <summary>I don't have pipx installed.</summary>
+
+ZMK CLI can be installed with pip, but using [pipx](https://github.com/pypa/pipx) is recommended to avoid conflicts between Python packages.
```

Comment:
I'm not sure this should be hidden inside the expander. Otherwise, if someone decides to go against the "(recommended)", they will likely not see that they can use pip instead.

Maybe we should also add something like "If you do not want to use pipx, replace `pipx` with `pip` in the rest of the instructions"?